jvm/src/main/scala/eval/ParallelInterpreter.scala

```diff
@@ -0,0 +1,65 @@
+package com.azavea.maml.eval
+
+import com.azavea.maml.ast._
+import com.azavea.maml.error._
+import com.azavea.maml.eval.directive._
+
+import cats._
+import cats.implicits._
+import cats.data.Validated._
+import cats.data.{NonEmptyList => NEL, _}
+import cats.effect.ContextShift
+
+import scala.reflect.ClassTag
+
+class ParallelInterpreter[F[_]: Monad, G[_]](directives: List[Directive])(
+    implicit Par: Parallel[F, G],
+    contextShift: ContextShift[F]
+) extends Interpreter[F] {
+  def apply(exp: Expression): F[Interpreted[Result]] = {
+    val children = evalInF(exp)
+    val out = children map {
+      _.andThen({ childRes =>
+        instructions(exp, childRes)
+      })
+    }
+    out
+  }
+
+  def evalInF(
+      expression: Expression
+  )(implicit contextShift: ContextShift[F]): F[Interpreted[List[Result]]] = {
+    val resultsF: F[List[Interpreted[Result]]] =
+      expression.children parTraverse { expr =>
```

Comment:
I'm genuinely a little bit shocked that this worked with so little fuss.

Reply:
I also was a little surprised, but it makes some sense -- `parTraverse` doesn't introduce the intermediate structure that `Concurrent` / `Fiber`s introduce, so :man_shrugging: